modules/pam_unix/pam_unix_acct.c

```diff
@@ -110,15 +110,10 @@ int _unix_run_verify_binary(pam_handle_t *pamh, unsigned long long ctrl,
       _exit(PAM_AUTHINFO_UNAVAIL);
     }
 
-    if (geteuid() == 0) {
-      /* must set the real uid to 0 so the helper will not error
-         out if pam is called from setuid binary (su, sudo...) */
-      if (setuid(0) == -1) {
-          pam_syslog(pamh, LOG_ERR, "setuid failed: %m");
-          printf("-1\n");
-          fflush(stdout);
-          _exit(PAM_AUTHINFO_UNAVAIL);
-      }
+    /* must set the real uid to 0 so the helper will not error
+       out if pam is called from setuid binary (su, sudo...) */
+    if (setuid(0) == -1) {
+        pam_syslog(pamh, LOG_DEBUG, "setuid failed: %m");
     }
```

Comment:
Shouldn't it still exit with an error if `geteuid() == 0`?

Reply:
I'd say that makes sense.

Reply:
OK, the next thing I'd suggest is to use `LOG_DEBUG` only for `geteuid() != 0` case. Let's keep `LOG_ERR` in case of exiting because it's indeed an error, and there is a long standing practice to log it this way.

Reply:
Yeah, let's keep old behavior for this case. Although I'm curious how to trigger it.

Reply:
> Yeah, let's keep old behavior for this case. Although I'm curious how to trigger it.

A seccomp filter that disallows it?

Reply:
Interesting idea! :)